### DISCUSSION TITLE
Unable to Create DigitalOcean Volume

### DISCUSSION THREAD

**Discussion by ShivamB25** [coder/coder#10159] [General] [answered]
issue while trying to create a DigitalOcean volume for the workspace.indicating authentication failure with DigitalOcean, preventing the creation of the volume.
**Steps to Reproduce:**
1. Run the `coder create` command with the specified configuration.
2. During the process, when creating a DigitalOcean volume, an authentication error occurs.

**Expected Behavior:**
The DigitalOcean volume should be created successfully as part of the workspace provisioning process without authentication errors.

**Actual Behavior:**
The creation of the DigitalOcean volume failed due to an authentication issue, and the workspace provisioning process was interrupted.

**Additional Information:**
- Terraform Version: 1.4.6
- DigitalOcean Token: doctl was authenticated + export DIGITALOCEAN_TOKEN= (was done after ensuring write permission was given)

**Error Message:**
```
2023-10-02 09:47:29.116+02:00 Error: Error creating Volume: POST https://api.digitalocean.com/v2/volumes: 401 (request "09e271cb-8298-49ef-b219-cd3445e8e60e") Unable to authenticate you
2023-10-02 09:47:29.116+02:00 on main.tf line 266, in resource "digitalocean_volume" "home_volume":
2023-10-02 09:47:29.117+02:00   266: resource "digitalocean_volume" "home_volume" {
```

**Comment by ShivamB25** [comment]
@matifali sir any bypass/workaround for this?

**Comment by matifali** [comment]
@ShivamB25, the issue looks like you have an authentication problem with the provider. This is possibly a bug with do provider. Have you searched their issues or filed an issue with them?

**Comment by ShivamB25** [comment]
@matifali i am using the api key with both doctl with exported as a variable too on the command line

**Comment by matifali** [comment]
The variable needs to be set where Coder is installed and should be accessible to the Coder user. This really depends on how Coder is run. 
If you are running coder as a system service, the place would be `/etc/coder.d/coder.env` if you are using the docker image, then it should be provided as an environment variable to the coder container.

**Comment by ShivamB25** [comment]
> The variable needs to be set where Coder is installed and should be accessible to the Coder user. This really depends how Coder is installed. If you are running coder as a system service the best place would be `/etc/coder.d/coder.env` if you are using the docker image then it should be provided as an environment variables to the coder container.

running as non-service / docker.
it is running a on a non-root user on ubuntu

**Comment by matifali** [comment]
@ShivamB25 try this https://coder.com/docs/v2/latest/install/packages

**Comment by ShivamB25** [comment]
@matifali i will try running it as a service and input those vars and give it a try again in a bit. thanks

**Comment by ShivamB25** [comment]
@matifali thanks it worked with service file not sure why wasn't it working with exported env and doctl already authenticated

**Reply by matifali** [reply, marked as answer]
Authenticating `doctl` CLI and authenticating the terraform provider are two different things. The `coder` user runs terraform, so the `coder` user needs access to this env variable. Glad that you fixed it.

**Reply by ShivamB25** [reply]
much thank for advising using service file instead
